apps/desktop-wallet/src/api/addressesFungibleTokensInfoDataHooks.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AddressHash } from '@alephium/shared'
+import { ALPH, TokenInfo } from '@alephium/token-list'
+import { explorer } from '@alephium/web3'
+import { useQueries } from '@tanstack/react-query'
+import { useMemo } from 'react'
+
+import { addressTokensBalanceQuery } from '@/api/addressQueries'
+import { useFungibleTokenList } from '@/api/fungibleTokenListDataHooks'
+import { useAppSelector } from '@/hooks/redux'
+import { selectAddressesLatestHash } from '@/storage/transactions/transactionsSelectors'
+
+export const useAddressesListedFungibleTokens = (addressHash?: AddressHash) => {
+  const { data: fungibleTokenList } = useFungibleTokenList()
+  const latestAddressesTxHashes = useAppSelector((s) => selectAddressesLatestHash(s, addressHash))
+  const networkId = useAppSelector((s) => s.network.settings.networkId)
+
+  const { data, isLoading } = useQueries({
+    queries: latestAddressesTxHashes.map(({ addressHash, latestTxHash }) =>
+      addressTokensBalanceQuery({ addressHash, latestTxHash, networkId })
+    ),
+    combine: (results) => ({
+      data: results.reduce(
+        (acc, { data }) => {
+          data?.map(({ tokenId }) => {
+            const listedFungibleToken = fungibleTokenList?.find((token) => token.id === tokenId)
+            const alreadyAddedToArray = acc.some((token) => token.id === listedFungibleToken?.id)
+
+            if (listedFungibleToken && !alreadyAddedToArray) acc.push(listedFungibleToken)
+          })
+          return acc
+        },
+        // Include ALPH in the results
+        [ALPH] as TokenInfo[]
+      ),
+      isLoading: results.some(({ isLoading }) => isLoading)
+    })
+  })
+
+  return {
+    data,
+    isLoading
+  }
+}
+
+export const useAddressesListedFungibleTokensWithPrice = (addressHash?: AddressHash) => {
```

Comment:
General comment: I'm not sure to understand how the hooks are split between those files. 
I'd rather have less specific files with more hooks in them, to get a better view of the API structure and of the data I can get. 

To take this one as exemple, it's not clear to me if this should be in here or in `addressesFungibleTokensPricesDataHooks.tsx`

Idea: we could maybe group hooks according to the type of data, similar to how the api is designed?..
`api/tokens/ft`
`api/tokens/nft`
`api/addresses/balances`
etc...

Reply:
still experimenting with structures. I wanted to write a message to you to present the current structure but I didn't have the time I needed last week and I put it in draft. I wanted to say something in line of "I wanna introduce this file called [...]DataHooks.tsx for hooks that mainly return `data` objects from Tanstack queries (see https://tanstack.com/query/latest/docs/framework/react/guides/render-optimizations#structural-sharing).

> To take this one as exemple, it's not clear to me if this should be in here or in addressesFungibleTokensPricesDataHooks.tsx

In this case, `addressesFungibleTokensPricesDataHooks.tsx` includes hooks that return the same thing. Prices. Hence the `[...]PricesDataHooks.tsx` suffix.

The `addressesFungibleTokensInfoDataHooks.tsx` includes hooks that return TokenInfo objects. Hence the `[...]TokensInfoDataHooks.tsx` suffix.

We can talk about it more tomorrow, I'll rework the PR and remove the draft status.